site/pages/api/delete-disruption.api.ts

```diff
@@ -0,0 +1,44 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { getPtSituationElementFromDraft } from "./publish.api";
+import { ERROR_PATH } from "../../constants";
+import { deletePublishedDisruption, getDisruptionById } from "../../data/dynamo";
+import { redirectTo, redirectToError } from "../../utils/apiUtils";
+import logger from "../../utils/logger";
+
+const deleteDisruption = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+    try {
+        const { query } = req;
+
+        const id = query?.id;
+
+        if (!id || Array.isArray(id)) {
+            throw new Error(
+                `Insufficient data provided for deleting a disruption by id: ${id ? id.toString() : "undefined"}`,
+            );
+        }
+
+        const disruption = await getDisruptionById(id);
+
+        if (!disruption) {
+            logger.error(`Disruption ${id} not found to delete`);
+            redirectTo(res, ERROR_PATH);
+            return;
+        }
+
+        await deletePublishedDisruption(getPtSituationElementFromDraft(disruption), id);
```

Comment:
We could probably simplify this by just passing through the disruption without converting it into a PtSituationElement since all we really need to delete it is the ID and the number of consequences

Reply:
done